src/support/thread_group.c

```diff
@@ -22,11 +22,21 @@ __wt_thread_run(void *arg)
 	thread = (WT_THREAD*)arg;
 	session = thread->session;
 
+	for (;;) {
+		if (!F_ISSET(thread, WT_THREAD_RUN))
+			return (WT_THREAD_RET_VALUE);
+		if (!F_ISSET(thread, WT_THREAD_ACTIVE))
+			__wt_cond_wait(session, thread->pause_cond,
+			    WT_THREAD_PAUSE * WT_MILLION, thread->chk_func);
```

Comment:
I strongly disagree with this.  Paused threads shouldn't even exist. I.e. this mechanism is replacing thread create/destroy.  We have customers who run 40 WT databases.  If most of those use the default eviction threads that is several hundred threads, most of which are paused.  There have been tickets where internal worker threads doing actual work waking 0.1 second was too much for that scenario and wasted a lot of CPU.  The algorithm is supposed to settle on a stable number so this is an experiment to see if keeping these threads existing is better than create/destroy.